src/test/optimizer/strategy/in_expression_rewrite_rule_test.cpp

```diff
@@ -80,34 +80,39 @@ TEST_F(InExpressionRewriteRuleTest, DisjunctionStrategy) {
     const auto input_lqp = PredicateNode::make(five_element_in_expression, node);
     const auto result_lqp = StrategyBaseTest::apply_rule(rule, input_lqp);
 
-    // clang-format off
-    // Account for different order produced by ExpressionUnorderedSet
-#ifdef __GLIBCXX__
-    const auto expected_lqp =
-      UnionNode::make(UnionMode::All,
-        UnionNode::make(UnionMode::All,
-          UnionNode::make(UnionMode::All,
-            UnionNode::make(UnionMode::All,
-              PredicateNode::make(equals_(col_a, 5), node),
-              PredicateNode::make(equals_(col_a, 3), node)),
-            PredicateNode::make(equals_(col_a, 2), node)),
-          PredicateNode::make(equals_(col_a, 4), node)),
-        PredicateNode::make(equals_(col_a, 1), node));
-#else
-    const auto expected_lqp =
-      UnionNode::make(UnionMode::All,
-        UnionNode::make(UnionMode::All,
-          UnionNode::make(UnionMode::All,
-            UnionNode::make(UnionMode::All,
-              PredicateNode::make(equals_(col_a, 5), node),
-              PredicateNode::make(equals_(col_a, 4), node)),
-            PredicateNode::make(equals_(col_a, 3), node)),
-          PredicateNode::make(equals_(col_a, 2), node)),
-        PredicateNode::make(equals_(col_a, 1), node));
-#endif
-    // clang-format on
-
-    EXPECT_LQP_EQ(result_lqp, expected_lqp);
+    // Can't use EXPECT_LQP_EQ here, because ExpressionUnorderedSet produces a non-deterministic order of predicates
+    auto values_found_in_predicates = std::vector<int>{};
+
+    // Checks that a given node is a predicate of the form `col_a = x` where x is an int and will be added to
+    // values_found_in_predicates
+    const auto verify_predicate_node = [&](const auto& node) {
+      EXPECT_EQ(node->type, LQPNodeType::Predicate);
+      auto predicate_node = std::dynamic_pointer_cast<PredicateNode>(node);
+      EXPECT_TRUE(predicate_node);
+      auto predicate = std::dynamic_pointer_cast<BinaryPredicateExpression>(predicate_node->predicate());
+      EXPECT_TRUE(predicate);
+      EXPECT_EQ(predicate->left_operand(), col_a);
+      EXPECT_EQ(predicate->right_operand()->type, ExpressionType::Value);
```

Comment:
ASSERT_EQ, otherwise the next line would be executed (and result in UB, likely?)

Reply:
Yepp. I honestly don't see too much of a value in `EXPECT_` anyway.